controllers/clusterresources/redisuser_controller.go

```diff
@@ -254,6 +267,55 @@ func (r *RedisUserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	return models.ExitReconcile, nil
 }
 
+func (r *RedisUserReconciler) detachUserFromDeletedCluster(
+	ctx context.Context,
+	clusterID string,
+	user *v1beta1.RedisUser,
+	logger logr.Logger,
+) error {
+
+	patch := user.NewPatch()
+	delete(user.Status.ClustersEvents, clusterID)
+
+	err := r.Status().Patch(ctx, user, patch)
+	if err != nil {
+		logger.Error(err, "Cannot detach clusterID from the Redis user resource",
+			"cluster ID", clusterID,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.PatchFailed,
+			"Detaching clusterID from the Redis user resource has been failed. Reason: %v",
+			err,
+		)
+		return err
+	}
+
+	controllerutil.RemoveFinalizer(user, user.DeletionUserFinalizer(clusterID, user.Namespace))
```

Comment:
`DeletionUserFinalizer` -> `DeletionFinalizer`

Reply:
It's a separate method for user,  DeletionFinalizer used for secrets